### DISCUSSION TITLE
Difference between branches and tags

### DISCUSSION THREAD

**Discussion by pszlazak** [eclipse-jgit/jgit#65] [Q&A] [answered]
Hello,
Either I'm doing something wrong or there is a difference between contents of Git branches and tags.  
I'm trying to check in which JGit version `bf70c9f4c` was added. For GitHub hosted repos I usually use GitHub UI:  
https://github.com/eclipse-jgit/jgit/commit/bf70c9f4c

![image](https://github.com/eclipse-jgit/jgit/assets/11884243/e79e0096-ed4a-431d-8caf-62c9e9cf74b7)

So the 1st tag where `bf70c9f4c` can be found is `v6.9.0.202402211805-m3`.

I see the same thing using CLI:
```bash
$ git tag --contains bf70c9f4c | sort -V
v6.9.0.202402211805-m3
v6.9.0.202402281855-rc1
v6.9.0.202403050045-r
v6.9.0.202403050737-r
v6.10.0.202405212237-m3
v6.10.0.202405282244-rc1
v6.10.0.202405290101-rc1
v6.10.0.202406032110-r
v6.10.0.202406032230-r
```

But when I check branches, I can see `bf70c9f4c` is available since 6.7:
```bash
$ git branch -a --contains bf70c9f4c | sort -V
  remotes/origin/HEAD -> origin/master
  remotes/origin/master
  remotes/origin/master%private
  remotes/origin/next
  remotes/origin/stable-6.7
  remotes/origin/stable-6.8
  remotes/origin/stable-6.9
  remotes/origin/stable-6.10
* master
```

I'm confused. Which one is valid?

**Comment by msohn** [comment, marked as answer]
This is both correct, looking at the version graph around that commit you can see that the commit was done on the stable-6.7 branch but no 6.7 version was tagged after this merge. The first tag after the commit was submitted is v6.9.0.202402211805-m3.
Here the version graph as displayed in EGit:

<img width="640" alt="Screenshot 2024-06-20 at 12 57 36" src="https://github.com/eclipse-jgit/jgit/assets/143355/107e3bef-4f16-4a90-aeea-0f15d9537bff">

**Reply by pszlazak** [reply]
Thank you @msohn! Wrong assumption from my side, that there will some `v6.7.xxxx` tag for any commit in stable-6.7 branch.

**Comment by msohn** [comment]
Most bug fixes are shipped in the next regular release with the next minor or major version. Only if there's a severe problem or security issue we create a service release on older releases.
